pkg/lib/cockpit-rsync-plugin.js

```diff
@@ -0,0 +1,31 @@
+const child_process = require("child_process");
+
+module.exports = class {
+    constructor(options) {
+        if (!options)
+            options = {};
+        this.dest = options.dest;
+        if (options.source) {
+            this.source = options.source;
+        } else {
+            this.source = "dist/";
+        }
+    }
+
+    apply(compiler) {
+        compiler.hooks.afterEmit.tapAsync('WebpackHookPlugin', (compilation, callback) => {
+            if (process.env.RSYNC) {
+                rsync(this.source, this.dest, process.env.RSYNC);
```

Comment:
rsync() is blocking. Is that right for tapAsync()? It feels to me it should rather use `child_process.spawn()` and hook up its `close` event to calling `callback`? I'm not very familiar with the webpack plugin API, though.

nodejs also offeers [promisify()](https://nodejs.org/api/util.html#utilpromisifyoriginal) to use it together with `tapPromise()`, which might be more convenient.